pkg/clients/rds/rds.go

```diff
@@ -452,7 +458,50 @@ func IsUpToDate(p v1beta1.RDSInstanceParameters, db rds.DBInstance) (bool, error
 		cmpopts.IgnoreFields(v1beta1.RDSInstanceParameters{}, "Tags"),
 		cmpopts.IgnoreFields(v1beta1.RDSInstanceParameters{}, "SkipFinalSnapshotBeforeDeletion"),
 		cmpopts.IgnoreFields(v1beta1.RDSInstanceParameters{}, "FinalDBSnapshotIdentifier"),
-	), nil
+		cmpopts.IgnoreFields(v1beta1.RDSInstanceParameters{}, "ApplyModificationsImmediately"),
+		cmpopts.IgnoreFields(v1beta1.RDSInstanceParameters{}, "AllowMajorVersionUpgrade"),
+		cmpopts.IgnoreFields(v1beta1.RDSInstanceParameters{}, "MasterPasswordSecretRef"),
+	) && !pwdChanged, nil
+}
+
+// GetPassword fetches the referenced input password for an RDSInstance CRD and determines whether it has changed or not
+func GetPassword(ctx context.Context, kube client.Client, r *v1beta1.RDSInstance) (string, bool, error) {
+	if kube == nil {
+		return "", false, nil
+	}
+	if r.Spec.ForProvider.MasterPasswordSecretRef == nil && r.Spec.WriteConnectionSecretToReference == nil {
+		return "", false, nil
+	}
+
+	var newPwd string
+	var pwdChanged bool
+	if r.Spec.ForProvider.MasterPasswordSecretRef != nil {
+		nn := types.NamespacedName{
+			Name:      r.Spec.ForProvider.MasterPasswordSecretRef.Name,
+			Namespace: r.Spec.ForProvider.MasterPasswordSecretRef.Namespace,
+		}
+		s := &corev1.Secret{}
+		if err := kube.Get(ctx, nn, s); err != nil {
+			return "", false, errors.Wrap(err, errGetPasswordSecretFailed)
+		}
+		newPwd = string(s.Data[r.Spec.ForProvider.MasterPasswordSecretRef.Key])
+	}
+
+	if r.Spec.WriteConnectionSecretToReference != nil {
+		nn := types.NamespacedName{
+			Name:      r.Spec.WriteConnectionSecretToReference.Name,
+			Namespace: r.Spec.WriteConnectionSecretToReference.Namespace,
+		}
+		s := &corev1.Secret{}
+		if err := kube.Get(ctx, nn, s); err != nil {
+			pwdChanged = newPwd != ""
+		} else {
+			outPwd := string(s.Data[v1alpha1.ResourceCredentialsSecretPasswordKey])
+			pwdChanged = newPwd != "" && outPwd != newPwd
+		}
```

Comment:
```suggestion
		if err := kube.Get(ctx, nn, s); resource.IgnoreNotFound(err) != nil {
			return "", false, err
		}
		pwdChanged = newPwd != string(s.Data[v1alpha1.ResourceCredentialsSecretPasswordKey])
		err = nil
		return
```

On `NotFound` case is OK to skip. For other errors, we need to return them. Also we don't need to check specifically against `""` because if it's `NotFound`, then `string(s.Data[v1alpha1.ResourceCredentialsSecretPasswordKey])` will be `""`.

Reply:
Great call

Reply:
I kept the `newPwd != ""` because I think we want to make sure that at least some value was pulled from the input secret, otherwise we may report that the password changed (a value in output but no value in input) but also send back an empty string for `newPwd`. This wouldn't matter in `Create` as we only care about the value of the password, but I'm unsure what the implications are of providing an empty password in `Update` and what having that included in the AWS request body would do